apis/hive/v1/clusterdeployment_types.go

```diff
@@ -401,6 +419,11 @@ const (
 	// AWSPrivateLinkFailedClusterDeploymentCondition is true controller fails to setup private link access
 	// for the cluster.
 	AWSPrivateLinkFailedClusterDeploymentCondition ClusterDeploymentConditionType = "AWSPrivateLinkFailed"
+
+	// These are conditions that are copied from ClusterInstall on to the ClusterDeployment object.
+	ClusterInstallFailedClusterDeploymentCondition          ClusterDeploymentConditionType = "ClusterInstallFailed"
+	ClusterInstallCompletedClusterDeploymentCondition       ClusterDeploymentConditionType = "ClusterInstallCompleted"
```

Comment:
I've got this as "Stopped" in my current suggestion comment in the enhancement, unless someone objects I plan to update enhancement for that. I like it as it feels useful in both Success/Failure scenarios, which is what I want here.

Reply:
> https://github.com/openshift/hive/pull/1356#discussion_r616618445

i'm not a fan of stopped. the completed and failed conditions already can provide users with that info. it's just duplicate info that requires additional handling imo

Reply:
Stopped is a replacement for Completed. I am worried about Completed because we need a state for install limits hit, and we gave up. That was going to be Installed=False, ProvisionFailed=True, and something to indicate we gave up. Completed=True there may easily confuse someone. Stopped I feel is less likely to.

So we're already committed to Failed="an attempt failed but we might still be retrying", Installed is obvious, how to we represent that we gave up and are no longer trying?

Reply:
if we look at the implementation conditions only, and set aside how they look on clusterdeployment,

Installed = true/false , are we installed
Failed = true/false , an attempt failed (we might be retrying to fix this error)
Stopped = true/false, are we trying to reach the Installed true

when we have succeed.
Installed = true
Failed = false
Stopped = true

when we installing but have previously failed
Installed = false
Failed = true
Stopped = false

when we have given up because of retries,
Installed = false
Failed = true
Stopped = true 

this condition set seems acceptable.

Reply:
For posterity after discussing as a team we ended up with Completed + Failed + Stopped. Will be updating enhancement soon.